packages/prompts/src/index.ts

```diff
@@ -995,3 +1024,113 @@ export const tasks = async (tasks: Task[], opts?: CommonOptions) => {
 		s.stop(result || task.title);
 	}
 };
+
+export interface TaskLogOptions extends CommonOptions {
+	title: string;
+	limit?: number;
+	spacing?: number;
+	retainLog?: boolean;
+}
+
+export interface TaskLogMessageOptions {
+	raw?: boolean;
+}
+
+export interface TaskLogCompletionOptions {
+	renderLog?: boolean;
+}
+
+/**
+ * Renders a log which clears on success and remains on failure
+ */
+export const taskLog = (opts: TaskLogOptions) => {
+	const output: Writable = opts.output ?? process.stdout;
+	const columns = getColumns(output);
+	const secondarySymbol = color.dim(S_BAR);
+	const spacing = opts.spacing ?? 1;
+	const barSize = 3;
+	const retainLog = opts.retainLog === true;
+
+	output.write(`${secondarySymbol}\n`);
+	output.write(`${color.green(S_STEP_SUBMIT)}  ${opts.title}\n`);
+	for (let i = 0; i < spacing; i++) {
+		output.write(`${secondarySymbol}\n`);
+	}
+
+	let buffer = '';
+	let fullBuffer = '';
+	let lastMessageWasRaw = false;
+
+	const clear = (clearTitle: boolean): void => {
+		if (buffer.length === 0) {
+			return;
+		}
+		const bufferHeight = buffer.split('\n').reduce((count, line) => {
+			if (line === '') {
+				return count + 1;
+			}
+			return count + Math.ceil((line.length + barSize) / columns);
+		}, 0);
+		const lines = bufferHeight + 1 + (clearTitle ? spacing + 2 : 0);
+		output.write(erase.lines(lines));
+	};
+	const renderBuffer = (): void => {
+		if (retainLog === true && fullBuffer.length > 0) {
+			log.message(`${fullBuffer}\n${buffer}`, {
+				output,
+				secondarySymbol,
+				symbol: secondarySymbol,
+				spacing,
+			});
+		} else {
+			log.message(buffer, { output, secondarySymbol, symbol: secondarySymbol, spacing });
+		}
+	};
+
+	return {
+		message(msg: string, mopts?: TaskLogMessageOptions) {
+			clear(false);
+			const dimMessage = color.dim(msg);
+			if ((mopts?.raw !== true || !lastMessageWasRaw) && buffer !== '') {
+				buffer += '\n';
+			}
+			buffer += dimMessage;
+			lastMessageWasRaw = mopts?.raw === true;
```

Comment:
coloring is covered by `print`
```suggestion
			clear(false);
			if ((mopts?.raw !== true || !lastMessageWasRaw) && buffer !== '') {
				buffer += '\n';
			}
			buffer += msg;
			lastMessageWasRaw = mopts?.raw === true;
```